nearcore/src/runtime/mod.rs

```diff
@@ -358,9 +349,12 @@ impl NightshadeRuntime {
         ShardUId { version: shard_version, shard_id: shard_id as u32 }
     }
 
-    pub fn get_shard_layout(&self, epoch_id: &EpochId) -> ShardLayout {
+    fn account_id_to_shard_uid(&self, account_id: &AccountId, epoch_id: &EpochId) -> ShardUId {
         let mut epoch_manager = self.epoch_manager.as_ref().write().expect(POISONED_LOCK_ERR);
-        epoch_manager.get_shard_layout(epoch_id).unwrap().clone()
+        let shard_layout = epoch_manager.get_shard_layout(epoch_id).unwrap();
```

Comment:
is it actually reasonable to unwrap here? How do we guarantee that we know the epoch?

Reply:
That's a good question. I didn't think very carefully about errors at all when I was designing the interface. I will go through these functions again and fix that